osu.Game/Scoring/ScoreModelManager.cs

```diff
@@ -63,6 +63,11 @@ protected override void Populate(ScoreInfo model, ArchiveReader? archive, Realm
             if (!model.Ruleset.IsManaged)
                 model.Ruleset = realm.Find<RulesetInfo>(model.Ruleset.ShortName);
 
+            // These properties are known to be non-null, but these final checks ensure a null hasn't come from somewhere (or the refetch has failed).
+            // Under no circumstance do we want these to be written to realm as null.
+            if (model.BeatmapInfo == null) throw new ArgumentNullException(nameof(model.BeatmapInfo));
+            if (model.Ruleset == null) throw new ArgumentNullException(nameof(model.Ruleset));
```

Comment:
In light of 744084b4, will these ever realistically trigger if someone doesn't explicitly try to break the model by inserting a `null`?

Reply:
they shouldn't no